### PR TITLE
Tutorial fermions

### PR DESCRIPTION
**Pull request by gcarleo**
Add a tutorial on spinless lattice fermions, Slater and Slater neural backflow

### CODE REVIEW

File: docs/tutorials/lattice-fermions.ipynb [hunk 1]
```diff
@@ -0,0 +1,762 @@
+{
```
Comment (PhilipVinc) [inline docs/tutorials/lattice-fermions.ipynb:1]: <p>gneral</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/netket/netket/pull/1581/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

Comment (PhilipVinc) [inline docs/tutorials/lattice-fermions.ipynb:1]: <p>Can you just add a line mentioning that <code>s=None</code> is used to say that they carry no Spin, and if one wanted to do spin-1/2 fermions one could just change <code>s=1/2</code>? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/netket/netket/pull/1581/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (PhilipVinc) [inline docs/tutorials/lattice-fermions.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.        log_det = jnp.linalg.slogdet(phi)
</pre><p>I would change it to</p><pre class="ql-syntax" spellcheck="false">&nbsp; &nbsp; sign, logabsdet = jnp.linalg.slogdet(phi)
&nbsp; &nbsp; return logabsdet.astype(complex) + jnp.log(sign.astype(complex))
</pre><p>which is more comprehensible (the second conversion to array is not needed)</p><p> </p><p>or even, the one that is slightly more correct because it preserves single-precision if ever used</p><pre class="ql-syntax" spellcheck="false">&nbsp; &nbsp; sign, logabsdet = jnp.linalg.slogdet(phi)
&nbsp; &nbsp; &nbsp; &nbsp; return jnp.log(sign.astype(nk.jax.dtype_complex(phi.dtype))) + logabsdet.astype(nk.jax.dtype_complex(phi.dtype)
</pre><p>but that might be more confusing.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/netket/netket/pull/1581/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (PhilipVinc) [inline docs/tutorials/lattice-fermions.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #20.                phi = kernel[:, jnp.where(n, size=self.hilbert.n_fermions)[0]]
</pre><p> </p><p>It took me a moment to figure out this line. I don't like it (but I'm not sure how to do it nicely). Maybe it's worth adding a comment saying that jnp where gives you the indices of the orbitals and then you index? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/netket/netket/pull/1581/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (PhilipVinc) [inline docs/tutorials/lattice-fermions.ipynb:1]: <p>Might be worth mentioning that if the user has scarce mathematical knowledge he can check if his ansatz is holomorphic, at least on a set of points, by running</p><p> </p><pre class="ql-syntax" spellcheck="false">nk.utils.is_probably_holomorphic(vstate._apply_fun, vstate.parameters, vstate.samples)
</pre><p> </p><p>which in this case will return False.</p><p> </p><p>By the way, for real parameters all the holomorphic logic is completely bypassed, and one could also avoid to specify it altogether..</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/netket/netket/pull/1581/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (PhilipVinc) [inline docs/tutorials/lattice-fermions.ipynb:1]: <p>no plots? :(</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/netket/netket/pull/1581/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='52'/>

Comment (gcarleo) [reply docs/tutorials/lattice-fermions.ipynb:1]: yes I am using now `nonzero`, hopefully this is clearer? by some arcane reason this is also faster than using where...

Comment (gcarleo) [reply docs/tutorials/lattice-fermions.ipynb:1]: good point, just removed holomorphic alltogether

Comment (PhilipVinc) [reply docs/tutorials/lattice-fermions.ipynb:1]: Much clearer now, thanks.
I suspect the speedup is because nonzero uses some masking tricks, while jnp.where uses a find-like kernel that is probably better suited for very large arrays but very bad for the small ones we use.

Comment (gcarleo) [reply docs/tutorials/lattice-fermions.ipynb:1]: <p>I have added a plot now, and also another wave function to be complete... I think we can merge</p>

**Review by PhilipVinc — APPROVED** [review]
